aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -337,11 +337,81 @@ Tensor& addbmm__cuda(Tensor& self,
 
 Tensor addbmm_cuda(const Tensor& self,
                    const Tensor& batch1, const Tensor& batch2,
-                   Scalar beta, Scalar alpha) 
+                   Scalar beta, Scalar alpha)
 {
   Tensor out = at::empty({0}, self.options());
   addbmm_out_cuda(out, self, batch1, batch2, beta, alpha);
   return out;
 }
 
+Tensor dot_cuda(const Tensor& self, const Tensor& other) {
+  at::NoNamesGuard guard;
+
+  TORCH_CHECK(
+      self.dim() == 1 && other.dim() == 1,
+      "1D tensors expected, got, ",
+      self.dim(),
+      ", ",
+      other.dim(),
+      " tensors");
+  TORCH_CHECK(
+      self.scalar_type() == other.scalar_type(),
+      "dot : expected both vectors to have same dtype, but found ",
+      self.scalar_type(),
+      " and ",
+      other.scalar_type());
+  TORCH_CHECK(
+      self.numel() == other.numel(),
+      "inconsistent tensor size, expected tensor [",
+      self.numel(),
+      "] and src [",
+      other.numel(),
+      "] to have the same number of elements, but got ",
+      self.numel(),
+      " and ",
+      other.numel(),
+      " elements respectively");
+  TORCH_CHECK(
+      self.device() == other.device(),
+      "expected all tensors to be on the same device. Found: ",
+      self.device(),
+      ", ",
+      other.device());
+
+  int n = static_cast<int>(self.numel());
+  int incx = static_cast<int>(self.stride(0));
+  int incy = static_cast<int>(other.stride(0));
+  if (n == 1) {
+    incx = 1;
+    incy = 1;
+  }
+  TORCH_CHECK(
+      (n <= INT_MAX) && (incx <= INT_MAX) && (incy <= INT_MAX),
```

Comment:
you have to first check that self.numel(), self.stride(0) and other.stride(0) (which are int64_t) fit in int type, and then convert. This check is guaranteed to pass, because n, incx and incy are int already.

Reply:
oops, good catch